pkg/uprobetracer/usdt.go

```diff
@@ -55,6 +121,94 @@ func alignUp[T int | int32 | int64 | uint | uint32 | uint64](n T, align T) T {
 	return (n + align - 1) / align * align
 }
 
+func encodeArgument(arg string) uint64 {
+	/**
+	 * support the following patterns:
+	 * <size>@%<reg>
+	 * <size>@<offset>(%<reg>)
+	 */
+
+	encodeField := func(encodedValue *uint64, fieldOffset uint, fieldLength uint, fieldValue uint64) {
+		*encodedValue |= (fieldValue & (1<<(fieldLength+1) - 1)) << fieldOffset
```

Comment:
This is encoded by ig and decoded in include/gadget/usdt_argument.h, which will be compiled into third-party gadgets. So this encoding becomes parts of the ABI between ig and the gadget, and therefore it would need to be documented. I am concerned how we are increasing the API surface, and how difficult it would be to make changes. ig version n+1 would need to support gadgets compiled  with include/gadget/usdt_argument.h from previous versions of ig.

I would like bpf extensions (BPF_PROG_TYPE_EXT, Linux >=5.6) to be used as a way to reduce the API surface between ig and gadgets. In this way, the API is reduced to the function `usdt_get_argument()`, and the ebpf maps (__usdt_args_buffer and __usdt_args_info) become implementation details of the ebpf extension, invisible to the gadget.

Reply:
cc @mauriciovasquezbernal